ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/castor/algorithm/CastorFullOptimization.java

```diff
@@ -521,13 +519,13 @@ private static class SecondPreventiveRaoResult {
         private final PerimeterResult perimeterResult;
         private final PrePerimeterResult postPraSensitivityAnalysisOutput;
         private final Set<RemedialAction<?>> remedialActionsExcluded;
-        private final AppliedRemedialActions appliedCras;
+        private final AppliedRemedialActions appliedArasAndCras;
 
-        public SecondPreventiveRaoResult(PerimeterResult perimeterResult, PrePerimeterResult postPraSensitivityAnalysisOutput, Set<RemedialAction<?>> remedialActionsExcluded, AppliedRemedialActions appliedCras) {
+        public SecondPreventiveRaoResult(PerimeterResult perimeterResult, PrePerimeterResult postPraSensitivityAnalysisOutput, Set<RemedialAction<?>> remedialActionsExcluded, AppliedRemedialActions appliedArasAndCras) {
             this.perimeterResult = perimeterResult;
             this.postPraSensitivityAnalysisOutput = postPraSensitivityAnalysisOutput;
             this.remedialActionsExcluded = remedialActionsExcluded;
-            this.appliedCras = appliedCras;
+            this.appliedArasAndCras = appliedArasAndCras;
```

Comment:
is there a more elegant way to do all this ? i find it hard to follow.
maybe remove this field and re-build the set everytime it is necessary instead of storing it temporarily then copying only some values ;
maybe add the Instant to addAppliedNetworkActionsPostContingency too, even if we call it twice...